plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts

```diff
@@ -122,24 +119,7 @@ export class DefaultCatalogCollatorFactory implements DocumentCollatorFactory {
       entitiesRetrieved += entities.length;
 
       for (const entity of entities) {
-        yield {
-          title: entity.metadata.title ?? entity.metadata.name,
-          location: this.applyArgsToFormat(this.locationTemplate, {
```

Comment:
Should this field still be under the control of the collator factory perhaps? Not sure that the transformer should ever be allowed to modify that; the fact that it's given as a constructor parameter to this class and doesn't have any sensible variations, plus looks a bit odd as a second arg to the transformer, makes it sound like it shouldn't be under the control of the transformer.

Reply:
Could be the same thing with authorization perhaps too? Not sure about that one. @iamEAP or @emmaindal may know more. Leave it inside the transformer for now

Reply:
+1 on this.  The `authorization` stuff should definitely not be modifiable, as it concerns permissions.

And regarding `location`, we already have a mechanism for modifying it (that's become something of a convention in collators), so it's probably best to keep the `locationTemplate` mechanism as the only way to modify the `location`, and specifically disallow modifying it via this proposed mechanism.

Reply:
done, now location and authorization create inside the factory